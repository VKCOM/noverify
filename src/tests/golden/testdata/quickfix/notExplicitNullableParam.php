<?php

function notNullableString(string $filed = null) {}

function notNullableCallable(callable $a = null) {}

class MyClass1 {}

class MyClass2 {
    public function myMethod(MyClass1 $a = null) {}
}

function nullableArray(array $a = null) {}

function multipleArgsExample(string $a, int $b = null, bool $c = null) {}

function nullableOrString(null|string $a = null) {}

function mixedParam(mixed $a = null) {}
