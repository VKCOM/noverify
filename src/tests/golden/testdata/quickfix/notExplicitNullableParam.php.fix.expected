<?php

function notNullableString(?string $filed = null) {}

function notNullableCallable(?callable $a = null) {}

class MyClass1 {}

class MyClass2 {
    public function myMethod(?MyClass1 $a = null) {}
}

function nullableArray(?array $a = null) {}

function multipleArgsExample(string $a, ?int $b = null, ?bool $c = null) {}

function nullableOrString(null|string $a = null) {}

function mixedParam(mixed $a = null) {}

class SomeClass {

    public function willReturnCallback($callback): self
    {
        return $this;
    }

 private function funWithCallback(array $participation_statuses_data) {
    $this
      ->willReturnCallback(static function(int $user_id, array $exact_statuses = [], bool $need_hidden = true, ?Date $finished_at = null) use ($participation_statuses_data) {
            return;
      });
  }
}
